src/common/NetworkPrivate.cpp

```diff
@@ -3,16 +3,115 @@
 #include "common/NetworkManager.hpp"
 #include "common/NetworkResult.hpp"
 #include "common/QLogging.hpp"
-#include "debug/AssertInGuiThread.hpp"
 #include "singletons/Paths.hpp"
 #include "util/DebugCount.hpp"
 #include "util/PostToThread.hpp"
 
+#include <magic_enum/magic_enum.hpp>
 #include <QCryptographicHash>
 #include <QFile>
 #include <QNetworkReply>
 #include <QtConcurrent>
 
+#ifndef NDEBUG
+#    define CHATTERINO_WARN_SLOW_HTTP
+#endif
+
+#ifdef CHATTERINO_WARN_SLOW_HTTP
+#    include <QElapsedTimer>
+
+#    ifdef NDEBUG
+constexpr qsizetype SLOW_HTTP_THRESHOLD = 30;
+#    else
+constexpr qsizetype SLOW_HTTP_THRESHOLD = 90;
+#    endif
+#endif
```

Comment:
I don't like having different behaviors on debug and release builds when it can be avoided. I believe this should be axed to only have a single threshold and not depend on `NDEBUG`.

Reply:
It's not ideal, but I think it's better to have separate thresholds, as debug builds are usually slower - often by more than 2x. Furthermore, you can expect less performance from debug builds.

Reply:
I think this is acceptable since it's a best-effort thing, so we're trying to get useful data where possible to just look into things. We can easily adjust these values going forward